app/controllers/Search.scala

```diff
@@ -48,4 +56,92 @@ object Search extends Controller {
       }
     }
   }
+
+  def openSearchDescription(dtype: String) = Action { implicit request =>
+    def templateUrl(ftype: String) = {
+      routes.Search.index.absoluteURL + "/results?q={searchTerms}&page={startPage?}&perpage={count?}" +
+      "&target=" + dtype + "&format=" + ftype
+    }
+    Ok(
+      <OpenSearchDescription xmlns="http://a9.com/-/spec/opensearch/1.1/">
```

Comment:
I think it's a tough call, and there are multiple issues in play (no pun intended) here:

First, there is the fact that scala has built-in affordances for XML manipulation that it doesn't for HTML, etc that makes XML wrangling easy in code. Not clear there's a win in readability, etc to push XML into a Twirl template - maybe the reverse.

Second, there is the design issue of clean separation of concerns that argues for separating controller and view code, and this pulls the other way - towards moving things into view-land.

Third, there is the question of strengths and weaknesses of Twirl as a template engine itself, where I think arguments can be made either way.

I think we can look at this more systematically when we build an API, where the notion of a 'view' becomes more general than a web page/XML document. We might want a notion of view that encompasses more the twirl, etc 

But all that's more than I wanted to tackle with this small feature....